src/pages/CommandView/form-data/get-submit-argument/get-submit-argument.ts

```diff
@@ -0,0 +1,342 @@
+    const instanceNames = model.instance_names as { instance_name: string }
+
+    if (instanceNames && 'instance_name' in instanceNames) {
+      return instanceNames
+    }
+    return null
+  }
+  return null
+}
+
+/**
+ * Extract the parameter arguments from the model
+ *
+ * @param model
+ * @returns A {parameters: <model parameters>} object
+ */
+const getParameters = (model: ObjectWithStringKeys) => {
+  if (model && 'parameters' in model) {
+    return {
+      parameters: model.parameters as ObjectWithStringKeys | EmptyObject,
+    }
+  }
+
+  return null
+}
+
+/**
+ * Extract the parameter arguments from the model with all Bytes parameter
+ * values stripped out
+ *
+ * @param model
+ * @param command
+ * @returns A {parameters: <model parameters>} object with no Bytes parameters
+ */
+const getBytesParameters = (
+  model: ObjectWithStringKeys,
+  command: AugmentedCommand,
+) => {
+  const theCommandsParameters = command.parameters
+
+  if (model && 'parameters' in model) {
+    const modelCopy = {
+      ...(model as {
+        parameters: ObjectWithStringKeys
+      }),
+    }
+    const { parameters: modelParameters } = modelCopy
+    let modelParametersCopy = { ...modelParameters } as ObjectWithStringKeys
+
+    for (const commandParameter of theCommandsParameters) {
+      // TODO: we only go down one level, this can be improved if it's found to
+      // be necessary
+      if (commandParameter.type === 'Bytes') {
+        const { key: commandKey } = commandParameter
+
+        if (commandKey in modelParametersCopy) {
+          const { [commandKey]: theValue, ...rest } = modelParametersCopy
+          modelParametersCopy = rest as ObjectWithStringKeys
+        }
+      } else if (commandParameter.type === 'Dictionary') {
+        const { key } = commandParameter
+
+        if (key in modelParametersCopy) {
+          let theDictionary = modelParametersCopy[key] as ObjectWithStringKeys
+
+          for (const subParameter of commandParameter.parameters) {
+            if (subParameter.type === 'Bytes') {
+              const { key: subParameterKey } = subParameter
+
+              if (subParameterKey in theDictionary) {
+                const { [subParameterKey]: theDictionaryValue, ...rest } =
+                  theDictionary
+                theDictionary = rest
+              }
+            }
+          }
+          modelParametersCopy = {
+            ...modelParametersCopy,
+            [key]: theDictionary,
+          }
+        }
+      }
+    }
+
+    return { parameters: modelParametersCopy }
+  }
+  return null
+}
+
+const extractRequestTemplate = (
+  model: ObjectWithStringKeys,
+  command: AugmentedCommand,
+  hasBytes = false,
+): RequestTemplate => {
+  const comment = getComment(model)
+  const instance = getInstance(model)
+  const parameters = hasBytes
+    ? getBytesParameters(model, command)
+    : getParameters(model)
+
+  if (instance === null) {
+    throw new Error(
```

Comment:
This is meant to make the program blow up. The scenario is that a plugin writer created something that is totally broken, the backend didn't catch it (not at all an unlikely proposition, unfortunately) and the bad data made it to us. So we want to fail hard with a runtime error. That said, this could, and probably should, be caught so that the UI itself can report the error rather than falling back only on `console.error`.

FYI, the garden connection form has snackbar reporting for both success and failure. I had hope to eventually factor that out into a global component for reuse. You might not have seen that unless you had set up the backend with a child connection.

If you look at the code, be warned this was some of the first react code I ever wrote. It's pretty doggy ;-)